test/e2e/instrumentation/events.go

```diff
@@ -0,0 +1,169 @@
+			},
+		},
+		Regarding: v1.ObjectReference{
+			Namespace: namespace,
+		},
+		EventTime:           someTime,
+		Note:                "This is " + name,
+		Action:              "Do",
+		Reason:              "Test",
+		Type:                "Normal",
+		ReportingController: "test-controller",
+		ReportingInstance:   "test-node",
+	}
+}
+
+func eventExistsInList(client typedeventsv1beta1.EventInterface, namespace, name string) bool {
+	eventsList, err := client.List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "testevent-constant=true",
+	})
+	framework.ExpectNoError(err, "failed to list events")
+
+	for _, val := range eventsList.Items {
+		if val.ObjectMeta.Name == name && val.ObjectMeta.Namespace == namespace {
+			return true
+		}
+	}
+	return false
+}
+
+var _ = common.SIGDescribe("Events API", func() {
+	f := framework.NewDefaultFramework("events")
+	var client typedeventsv1beta1.EventInterface
+	var clientAllNamespaces typedeventsv1beta1.EventInterface
+
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet.EventsV1beta1().Events(f.Namespace.Name)
+		clientAllNamespaces = f.ClientSet.EventsV1beta1().Events(metav1.NamespaceAll)
+	})
+
+	/*
+		Release : v1.19
+		Testname: Event resource lifecycle
+		Description: Create an event, the event MUST exist.
+		The event is patched with a new note, the check MUST have the update note.
+		The event is deleted and MUST NOT show up when listing all events.
+	*/
+	ginkgo.It("should ensure that an event can be fetched, patched, deleted, and listed", func() {
+		eventName := "event-test"
+		testEvent := newTestEvent(f.Namespace.Name, eventName, "testevent-constant")
+
+		ginkgo.By("creating a test event")
+		_, err := client.Create(context.TODO(), testEvent, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create test event")
+
+		ginkgo.By("listing events in all namespaces")
+		foundCreatedEvent := eventExistsInList(clientAllNamespaces, f.Namespace.Name, eventName)
+		framework.ExpectEqual(foundCreatedEvent, true, "failed to find test event in list with cluster scope")
+
+		ginkgo.By("listing events in test namespace")
+		foundCreatedEvent = eventExistsInList(client, f.Namespace.Name, eventName)
+		framework.ExpectEqual(foundCreatedEvent, true, "failed to find test event in list with namespace scope")
+
+		ginkgo.By("patching the test event")
+		eventPatchNote := "This is a test event - patched"
+		eventPatch, err := json.Marshal(map[string]interface{}{
+			"note": eventPatchNote,
+		})
+		framework.ExpectNoError(err, "failed to marshal the patch JSON payload")
+
+		_, err = client.Patch(context.TODO(), eventName, types.StrategicMergePatchType, []byte(eventPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch the test event")
+
+		ginkgo.By("getting the test event")
+		event, err := client.Get(context.TODO(), eventName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to get test event")
+		event.ObjectMeta = testEvent.ObjectMeta
```

Comment:
@wojtek-t 
Looks like `client.Create` will add a lot of info into ObjectMeta, which will cause `DeepEqual` fail. So here I skip the comparison of the whole ObjectMeta. I'm not sure if this is the best way to do it. Or maybe it's better to get the event after `client.Create` and compare that with patched event?

Reply:
Emm I changed my mind - I think it might be better to include ObjectMeta, so I get `testEvent` after creation. Hope this can pass the test. Please correct me if I'm wrong.